src/SFML/Window/Unix/WindowImplX11.cpp

```diff
@@ -2139,4 +2154,28 @@ Vector2i WindowImplX11::getPrimaryMonitorPosition()
     return monitorPosition;
 }
 
+
+////////////////////////////////////////////////////////////
+void WindowImplX11::setWindowSizeConstraints() const
```

Comment:
I think this function should not do anything for a window that is not resizable, i.e. created without the flag sf::Style::Resize. Otherwise it can turn the window into a resizable one.

This can be done like so:
```c++
    if(m_useSizeHints)
        return;
```

Reply:
I'm not sure about this one. You currently can make the window not resizable but still call `setSize` on it. Or do the limits not consider `setSize` calls?

Reply:
WindowBase.cpp:setSize implementation takes care of this situation by clamping. 

If the user sets a size for the x dimension at 50, but they previously set the lower limit to 200, it will clamp their own setSize to 200.